File: core/src/main/java/com/ustadmobile/core/view/HolidayCalendarDetailView.java
package com.ustadmobile.core.view;


import com.ustadmobile.core.db.UmProvider;
import com.ustadmobile.lib.db.entities.DateRange;
import com.ustadmobile.lib.db.entities.UMCalendar;

/**
 * Core View. Screen is for HolidayCalendarDetail's View
 */
public interface HolidayCalendarDetailView extends UstadView {


    // This defines the view name that is an argument value in the go() in impl.
    String VIEW_NAME = "HolidayCalendarDetail";

    //Any argument keys:
    String ARG_CALENDAR_UID = "CalendarUid";

    void setListProvider(UmProvider<DateRange> umProvider);

    void updateCalendarOnView(UMCalendar calendar);

    /**
     * Method to finish the screen / view.
     */
    void finish();


}

File: core/src/commonMain/kotlin/com/ustadmobile/core/view/HolidayCalendarDetailView.kt
package com.ustadmobile.core.view


import androidx.paging.DataSource
import com.ustadmobile.core.db.UmProvider
import com.ustadmobile.lib.db.entities.DateRange
import com.ustadmobile.lib.db.entities.UMCalendar

/**
 * Core View. Screen is for HolidayCalendarDetail's View
 */
interface HolidayCalendarDetailView : UstadView {

    fun setListProvider(umProvider: DataSource.Factory<Int, DateRange>)

    fun updateCalendarOnView(calendar: UMCalendar)

    /**
     * Method to finish the screen / view.
     */
    fun finish()

    companion object {


        // This defines the view name that is an argument value in the go() in impl.
        val VIEW_NAME = "HolidayCalendarDetail"

        //Any argument keys:
        val ARG_CALENDAR_UID = "CalendarUid"
    }


}

